### PR TITLE
Updates for .NET November 2024 Servicing and .NET 9 GA

### PR DESCRIPTION
**Pull request by lbussell**
See [Tasks needed for .NET 9 GA (#5059)](https://github.com/dotnet/dotnet-docker/issues/5059)

### CODE REVIEW

**Comment by lbussell** [comment]
@NikolaMilosavljevic I gave it my best effort porting the internal testing changes to this branch (which is based off of `main`). I added you as a reviewer so you can double-check it. If we start testing internal builds against this branch, I think we would iron out any issues with the merge pretty quickly.

File: README.samples.md [hunk 1]
```diff
@@ -80,40 +86,40 @@ The [Image Variants documentation](https://github.com/dotnet/dotnet-docker/blob/
...
+dotnetapp-chiseled-9.0 | [Dockerfile](samples/dotnetapp/Dockerfile.chiseled) | Ubuntu
+aspnetapp-9.0 | [Dockerfile](samples/aspnetapp/Dockerfile.alpine) | Alpine
+aspnetapp-chiseled-9.0 | [Dockerfile](samples/aspnetapp/Dockerfile.chiseled) | Ubuntu
+dotnetapp-8.0, dotnetapp, latest | [Dockerfile](samples/8.0/dotnetapp/Dockerfile.alpine) | Alpine
+dotnetapp-chiseled-8.0, dotnetapp-chiseled | [Dockerfile](samples/8.0/dotnetapp/Dockerfile.chiseled) | Ubuntu
+aspnetapp-8.0, aspnetapp | [Dockerfile](samples/8.0/aspnetapp/Dockerfile.alpine) | Alpine
+aspnetapp-chiseled-8.0, aspnetapp-chiseled | [Dockerfile](samples/8.0/aspnetapp/Dockerfile.chiseled) | Ubuntu
 
-### Linux arm32 Tags
+### Linux arm64 Tags
```
[the remark this answers is not in the document]
Comment (lbussell) [reply README.samples.md:98]: Makes sense. I'm not sure how we can address this without making changes to ImageBuilder. Should we file an issue and address this later?

[the remark this answers is not in the document]
Comment (mthalman) [reply README.samples.md:98]: I agree it requires an Image Builder change. Yes, I'm fine with having an issue logged as follow-up.

**Review by mthalman — APPROVED** [review]

**Comment by jander-msft** [comment]
Please move the .NET Monitor `latest` tags to the 9.0 images e.g. https://github.com/dotnet/dotnet-docker/pull/6042/commits/1363a995fe42fe30b2179fe018f0865b95898d8d

**Comment by lbussell** [comment]
Rebased to pick up the following commits and keep the version changes as the last commits:
- [Re-enable pre-build validation for internal servicing testing](https://github.com/dotnet/dotnet-docker/pull/5983/commits/a1e1e04d82916f0616effe5ff0f0aa4397a18623)
- [Update common Docker engineering infrastructure with latest](https://github.com/dotnet/dotnet-docker/pull/5983/commits/36f860ce263f6db9a82c220b55f0029e4240a7eb)

**Comment by lbussell** [comment]
Final internal validation run for .NET 9: https://dev.azure.com/dnceng/internal/_build/results?buildId=2580014&view=results (internal link)

**Review by mthalman — APPROVED** [review]

File: README.samples.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lbussell) [reply README.samples.md:98]: This is tracked by https://github.com/dotnet/docker-tools/issues/1497

**Pull request merged**